server/src/main/java/org/opensearch/cluster/metadata/Metadata.java

```diff
@@ -291,12 +291,12 @@ static Custom fromXContent(XContentParser parser, String name) throws IOExceptio
         final Map<String, IndexMetadata> indices,
         final Map<String, IndexTemplateMetadata> templates,
         final Map<String, Custom> customs,
-        String[] allIndices,
-        String[] visibleIndices,
-        String[] allOpenIndices,
-        String[] visibleOpenIndices,
-        String[] allClosedIndices,
-        String[] visibleClosedIndices,
+        List<String> allIndices,
```

Comment:
What is this change intended for? Can you update the PR description and may be raise a logical cut of the change in the PR if it is a part of a feature?